Src/FluentAssertions.Json/JTokenDifferentiator.cs

```diff
@@ -191,31 +199,36 @@ private static Difference FindJObjectDifference(JObject actual, JToken expected,
                 return new Difference(DifferenceKind.OtherName, path);
             }
             
-            return FindFirstDifference(actualProperty.Value, expectedProperty.Value, path, ignoreExtraProperties);
+            return FindFirstDifference(actualProperty.Value, expectedProperty.Value, path, ignoreExtraProperties, config);
         }
 
-        private static Difference FindValueDifference(JValue actualValue, JToken expected, JPath path)
+        private static Difference FindValueDifference(JValue actualValue, JToken expected, JPath path, Func<IJsonAssertionOptions<object>, IJsonAssertionOptions<object>> config = null)
         {
             if (!(expected is JValue expectedValue))
             {
                 return new Difference(DifferenceKind.OtherType, path, Describe(actualValue.Type), Describe(expected.Type));
             }
             
-            return CompareValues(actualValue, expectedValue, path);
+            return CompareValues(actualValue, expectedValue, path, config);
         }
 
-        private static Difference CompareValues(JValue actual, JValue expected, JPath path)
+        private static Difference CompareValues(JValue actual, JValue expected, JPath path, Func<IJsonAssertionOptions<object>, IJsonAssertionOptions<object>> config = null)
         {
             if (actual.Type != expected.Type)
             {
                 return new Difference(DifferenceKind.OtherType, path, Describe(actual.Type), Describe(expected.Type));
             }
-            
-            if (!actual.Equals(expected))
+
+            bool hasMismatches;
+            using (var scope = new AssertionScope())
+            {
+                actual.Value.Should().BeEquivalentTo(expected.Value, options => (JsonAssertionOptions<object>)config?.Invoke(new JsonAssertionOptions<object>())??options);
```

Comment:
This should apply `config` on top of the existing `options` instead of overwriting it.
Something like `config.Invoke(options)` (I'm aware that it does not compile)

Here's an example that fails, where `options => options` should apply on top of the globally defined assertions options.
```cs
AssertionOptions.AssertEquivalencyUsing(e => e
    .Using<double>(ctx => ctx.Subject.Should().BeApproximately(ctx.Expectation, 0.1))
    .WhenTypeIs<double>()
);

// Arrange
var actual = JToken.Parse("{ \"id\": 1.1232 }");
var expected = JToken.Parse("{ \"id\": 1.1235 }");


// Act & Assert
actual.Should().BeEquivalentTo(expected, options => options);
```